src/Visitors/SetPropertyValue.php

```diff
@@ -35,110 +24,36 @@ public function __construct(
         $this->typeIdentifier = new Identifier($propertyType);
     }
 
-    public function enterNode(Node $node): Node
+    protected function shouldUpdateNode(Node $node): bool
     {
-        if ($node instanceof Property && $node->getAttribute('parent') instanceof Class_ && $node->props[0]->name->name === $this->name) {
-            $this->updateProperty($node);
-            $this->isPropertyExists = true;
-        }
-
-        return $node;
-    }
-
-    public function leaveNode(Node $node): Node
-    {
-        if ($node instanceof Class_ && !$this->isPropertyExists) {
-            $node->stmts[] = new Property(
-                flags: ($this->accessModifier ?? AccessModifierEnum::Public)->value,
-                props: [$this->propertyItem],
-                type: $this->typeIdentifier,
-            );
-
-            return $this->rebuildClass($node);
-        }
-
-        return $node;
+        return $node instanceof Property
+            && $node->getAttribute('parent') instanceof Class_
+            && $this->name === $node->props[0]->name->name;
     }
 
-    protected function updateProperty(Property $property): void
+    protected function updateNode(Node $node): void
```

Comment:
could we set type for node?


```suggestion
    protected function updateNode(Property $node): void
```